### PR TITLE
Feature/analytics

### PR DESCRIPTION
**Pull request by paynemiller92**
### Overview of Changes

1. Integrated `Google Analytics v4` as a dependency, and configured the build to include our ID.
2. Created a new manager class, `Analytics Manager` that handles all calls to Google Analytics.
3. Added calls to send Analytics calls to AnalyticsManager from every visible Fragment and Activity.

### Strategy

I tried to design this in a way that was intutive and was easy for you to make additions to. Ideally, you can just add more methods for different analytics you wish to capture (i.e. something besides an event or screen view)

### Exampes of Analytic Calls

| Metric      | Example                                                                      |
|-------------|------------------------------------------------------------------------------|
| Screen View | ```AnalyticsManager.trackScreenView("About Screen");```                      |
| Click Event | ```AnalyticsManager.trackEvent("Home Screen", "Settings Tab Clicked", "Click");``` |

### CODE REVIEW

File: app/src/com/vlille/checker/ui/fragment/MapFragment.java [hunk 1]
```diff
@@ -160,6 +162,12 @@ public void onRationaleDenied(int requestCode) {
         Log.d(TAG, "onRationaleDenied:" + requestCode);
     }
 
+    @Override
+    public void onStop() {
```
Comment (ojacquemart) [inline app/src/com/vlille/checker/ui/fragment/MapFragment.java]: It should be `onStart`, no? 😉

**Comment by ojacquemart** [comment]
Thansk! Good job :)

I do not know how GA works for mobile apps.
Is there a way to stop the tracking actions? I closed the app and a few minutes later, GA tells me that there is still an active user.

**Comment by paynemiller92** [comment]
@ojacquemart I updated the `onStop()` call to `onStart()` LOL autocorrect got me there. I researched hte active user issue and it seems that the Active Users number is actually the number of users active within the past 30 days! See this l[ink](https://www.optimizesmart.com/understanding-active-users-report-google-analytics/).

File: app/src/com/vlille/checker/ui/HomeActivity.java [hunk 2]
```diff
@@ -130,9 +133,11 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
                 break;
             case R.id.main_menu_update_stations:
+                AnalyticsManager.trackEvent("Home Screen", "Updatae Stations Button Clicked", "Click");
```
Comment (ojacquemart) [inline app/src/com/vlille/checker/ui/HomeActivity.java]: Small typo here "Update"

Comment (paynemiller92) [reply app/src/com/vlille/checker/ui/HomeActivity.java]: @ojacquemart fixed!

**Comment by ojacquemart** [comment]
Ok, the active number now displays 0.

Could you squash your commits after the typo fix?
I am using git karma. You could take a look at: http://karma-runner.github.io/3.0/dev/git-commit-msg.html

**Comment by paynemiller92** [comment]
@ojacquemart sure thing. I will even update the README here in a bit to make using git-karma part of the `Contribution Guide`.

**Comment by paynemiller92** [comment]
@ojacquemart updated the commit. Let me know if the format is correct!

**Comment by ojacquemart** [comment]
LGTM :+1:

Thanks for your contribution. Hacktoberfest rocks!

**Pull request merged**